client/components/platform-checkout/save-user/phone-number-input.js

```diff
@@ -1,144 +1,111 @@
 /**
  * External dependencies
  */
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useRef } from 'react';
 import { __ } from '@wordpress/i18n';
 import intlTelInput from 'intl-tel-input';
+import './style.scss';
 
 /**
  * Internal dependencies
  */
 // eslint-disable-next-line import/no-unresolved
 import utils from 'iti/utils';
 
-const PhoneNumberInput = ( { handlePhoneNumberChange } ) => {
-	const [ inputValue, setInputValue ] = useState(
-		document.getElementById( 'billing_phone' )?.value ?? ''
-	);
+const PhoneNumberInput = ( {
+	onValueChange,
+	value,
+	onValidationChange = () => {},
+	inputProps = {},
+} ) => {
 	const [ inputInstance, setInputInstance ] = useState( null );
 	const [ isValid, setIsValid ] = useState( true );
+	const inputRef = useRef();
 
-	const handlePhoneNumberInputChange = ( e ) => {
-		setInputValue( e.target.value );
+	const handlePhoneNumberInputChange = () => {
 		if ( inputInstance ) {
-			handlePhoneNumberChange( inputInstance.getNumber() );
+			onValueChange( inputInstance.getNumber() );
+			onValidationChange( inputInstance.isValidNumber() );
 		}
 	};
 
-	const handlePhoneNumberValidation = () => {
+	const removeInternationalPrefix = ( phone ) => {
 		if ( inputInstance ) {
-			setIsValid( inputInstance.isValidNumber() );
-		} else {
-			setIsValid( true );
-		}
-	};
-
-	const ErrorText = () => {
-		if ( ! isValid ) {
-			const message = __(
-				'Please enter a valid mobile phone number.',
-				'woocommerce-payments'
+			return phone.replace(
+				'+' + inputInstance.getSelectedCountryData().dialCode,
+				''
 			);
-
-			return <p className="error-text">{ message }</p>;
 		}
-		return '';
-	};
-
-	useEffect( () => {
-		// const formSubmitButton = document.getElementById( 'place_order' );
-		const formSubmitButton = document.querySelector(
-			'form.woocommerce-checkout button[type="submit"]'
-		);
-
-		if ( ! formSubmitButton ) {
-			return;
-		}
-
-		const updateFormSubmitButton = () => {
-			if ( isValid ) {
-				formSubmitButton.removeAttribute( 'disabled' );
-			} else {
-				formSubmitButton.setAttribute( 'disabled', 'disabled' );
-			}
-		};
-
-		updateFormSubmitButton();
 
-		return () => {
-			// Clean up
-			formSubmitButton.removeAttribute( 'disabled' );
-		};
-	}, [ isValid ] );
+		return value;
+	};
 
 	useEffect( () => {
 		let iti = null;
-		const input = document.querySelector(
-			`input[aria-label="${ __(
-				'Mobile phone number',
-				'woocommerce-payments'
-			) }"]`
-		);
 
 		const handleCountryChange = () => {
-			handlePhoneNumberChange( iti.getNumber() );
+			onValueChange( iti.getNumber() );
```

Comment:
Let's call the `onValidationChange( iti.isValidNumber() )` also here. Currently if I change the country from the error message does not appear even if it is an invalid phone number for the newly selected country.